activerecord/lib/active_record/connection_adapters/mysql2_adapter.rb

```diff
@@ -20,7 +20,7 @@ def mysql2_connection(config)
       ConnectionAdapters::Mysql2Adapter.new(client, logger, options, config)
     rescue Mysql2::Error => error
       if error.message.include?("Unknown database")
-        raise ActiveRecord::NoDatabaseError.new(error.message, error)
+        raise ActiveRecord::NoDatabaseError.new(error.message)
```

Comment:
one more, `#cause` returns `nil` here. We can also do the below, but it doesn't guarantee that `ActiveRecord::NoDatabaseError` will always have the original error message.

```ruby
raise ActiveRecord::NoDatabaseError, error.message
```